### ISSUE TITLE
[naming-convention] Selector or modifier for global variable. (differ local / global)

### ISSUE THREAD

**Issue by infacto** [typescript-eslint/typescript-eslint#2608]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

```JS
{
  "rules": {
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "selector": "default",
        "format": ["strictCamelCase"]
      },
      {
        "selector": "variable",
        "modifiers": ["const"],
        // I want only global variables be allowed UPPER_CASE. All other variables camelCase.
        "format": ["strictCamelCase", "UPPER_CASE"]
      }
      {
        "selector": "property",
        "format": ["strictCamelCase"],
        "leadingUnderscore": "allow"
      },
      {
        "selector": "property",
        "modifiers": ["static", "readonly"],
        "format": ["strictCamelCase", "UPPER_CASE"]
      },
      {
        "selector": "typeLike",
        "format": ["StrictPascalCase"]
      }
    ]
  }
}
```

```TS
// This is allowed.
const MY_VALUE = 42;

class MyClass {
  myMethod() {
    // This is NOT allowed.
    const ANOTHER_VALUE = 27;

    // This is allowed.
    const anotherValue = 13;
  }
}
```

**Expected Result**

An option to select the global variable in a file.

**Actual Result**

I have no idea how to select global and local variables.

**Additional Info**

I use vscode eslint extension for viewing the result.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.2.0` |
| `@typescript-eslint/parser`        | `4.2.0` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `10.22.0` |

**Comment by glen-84** [comment]
Duplicate of #2318.
